### PR TITLE
fetch container health status via commit id instead of env name

### PR DESCRIPTION
**Pull request by liyaqin1**
the current method of getting container’s health doesn’t work for enovy because their container’s name are different from the env name

### CODE REVIEW

**Comment by liyaqin1** [comment]
tested on my devapp and looks good:
Screenshot 2023-10-06 at 5 04 33 PM

**Comment by liyaqin1** [comment]
Tested on dev-yaqinli and sent the data to dev1. Looks good:
```
2023-10-18 23:27:45,714 urllib3.connectionpool:1018 DEBUG Starting new HTTPS connection (1): teletraan-dev1-agent.pinadmin.com:443
2023-10-18 23:27:45,855 urllib3.connectionpool:482 DEBUG https://teletraan-dev1-agent.pinadmin.com:443 "POST /v1/system/ping HTTP/1.1" 200 294
2023-10-18 23:27:45,856 deployd.client.client:224 DEBUG PingRequest(hostId=i-0dd4b6ce5062c36f0, hostName=dev-yaqinli, hostIp=10.9.215.252, agentVersion=1.2.46-pinterest-1.0.2, autoscalingGroup=None, availabilityZone=us-east-1d, stageType=None, groups=['singer-prod-sidecar', 'envoy-devapp', 'tcollector-prod-sidecar', 'mcrouter-devapp', 'mcrouter-prod-sidecar', 'metrics-agent-prod-sidecar', 'zk_update_monitor-prod-sidecar'], accountId=998131032990, reports=PingReport(deployId=ODNR_A12T_G49nP8ENt2Ag, envId=nS2GIXkgQnK4pR9qG5ZpFg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=8M6Sgx4SRnaFC2rwhxfjBw, envId=8ss-6DLmQeWe7EMejOVWEw, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=IS_sXYi4RyGfHAojyXllcw, envId=inbDfs6KQICpRRfuKQw_bQ, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=envoy-A:healthy;envoy-endpoint-register:healthy;),PingReport(deployId=0N3ZYZxtRR6_W-ju2cg8Zg, envId=PWfsL2iYSwaQpB70hui5AA, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=3zZuGjXbReqgsgwkAFcQPQ, envId=2XwshUOHR36DGEAH4PzBwg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=mcrouter:healthy;),PingReport(deployId=adIhQO7lTgS9eyfD5XT20A, envId=kwUx6aHWQo6R174lT-eawQ, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=fDXKA0UMQYaV07g0TYYHwA, envId=typrFRIFR6-lDyaiNjAk9w, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=rXqFmaubTIGROjX0XT4VAg, envId=gu2DEmL4Tv2XOqDKFc7Axw, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=ratelimit-A:healthy;),PingReport(deployId=DcdD-aipRTKp4hmtmXg45g, envId=7ib3xSHSR3unnLdN-ViGxw, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=E8U5leLLSmmmAtkiqqy-kQ, envId=pZwvW9faQTW7vD19ER72Gg, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=tcollector:healthy;),PingReport(deployId=lIq9RH-eS3miHvcUxURWiA, envId=99xVHd7bSr608HSHqrCGbA, deployStage=POST_RESTART, status=TOO_MANY_RETRY, errorCode=1, errorMessage=None, failCount=3, extraInfo=None, deployAlias=None, containerHealthStatus=None),PingReport(deployId=qXyg0uh1RpeHfUiTHo_Q8g, envId=tLWOA1CsQ4W2Nu_kMFyYVA, deployStage=SERVING_BUILD, status=SUCCEEDED, errorCode=0, errorMessage=None, failCount=0, extraInfo=None, deployAlias=None, containerHealthStatus=None)) -> PingResponse(opCode=DELETE, deployGoal=DeployGoal(deployId=3zZuGjXbReqgsgwkAFcQPQ, envId=2XwshUOHR36DGEAH4PzBwg, envName=None, stageName=None, deployStage=SERVING_BUILD, build=None, deployAlias=None, agentConfig=None,scriptVariables=None, firstDeploy=False, isDocker=None))

```

File: deploy-agent/deployd/common/utils.py [hunk 1]
```diff
@@ -219,18 +219,34 @@ def get_info_from_facter(keys):
...
-
-def get_container_health_info(key):
+    
+def get_container_health_info(commit):
     try:
-        log.info(f"Get health info for container {key}")
-        cmd = ['docker', 'inspect', '-f', '{{.State.Health.Status}}', key]
+        log.info(f"Get health info for container with commit {commit}")
+        result = ""
+        cmd = ['docker', 'ps', '--format', '{{.Image}};{{.Names}}']
```
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:227]: can you add UT for this method?

File: deploy-agent/deployd/common/utils.py [hunk 2]
```diff
@@ -219,18 +219,34 @@ def get_info_from_facter(keys):
...
+                    parts = line.split(';')
+                    name = parts[1]
+                    try:
+                        command = ['docker', 'inspect', '-f', '{{.State.Health.Status}}', name]
+                        status = subprocess.run(command, check=True, stdout=subprocess.PIPE).stdout
+                        if status:
+                            result = result + name + ":" + status.decode().strip() + ";"
+                    except:
+                        continue
+            if result != "":
```
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:242]: return result if result else None

File: deploy-agent/deployd/common/utils.py (hunk 1 above)
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:227]: can youi include in the PR description the example output of docker ps cmd and docker inspect -f example?

File: deploy-agent/deployd/common/utils.py (hunk 2 above)
Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:242]: sure, thanks

File: deploy-agent/deployd/common/utils.py (hunk 1 above)
Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:227]: I once added some logs in the method. Here is the example:
```
2023-10-18 23:27:41,145 deployd.common.utils:241 INFO Get health info for container with commit c0e2ee123f1fc442a84f94c92263d05c4a29d193
2023-10-18 23:27:41,167 deployd.common.utils:249 INFO for commit c0e2ee123f1fc442a84f94c92263d05c4a29d193: 998131032990.dkr.ecr.us-east-1.amazonaws.com/beacon:c0e2ee123f1fc442a84f94c92263d05c4a29d193;beacon
2023-10-18 23:27:41,183 deployd.common.utils:241 INFO Get health info for container with commit d74918918ff41e3877e0e91a996f049f87b7f75c
2023-10-18 23:27:41,203 deployd.common.utils:241 INFO Get health info for container with commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32
2023-10-18 23:27:41,225 deployd.common.utils:249 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: 998131032990.dkr.ecr.us-east-1.amazonaws.com/sox-traffic-envoy:86d3a57a51f62aa5a03cb66e6c64812e5e386e32;envoy-A
2023-10-18 23:27:41,242 deployd.common.utils:255 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: name is envoy-A
2023-10-18 23:27:41,242 deployd.common.utils:257 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: name is envoy-A, status is b'healthy\n'
2023-10-18 23:27:41,242 deployd.common.utils:249 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: 998131032990.dkr.ecr.us-east-1.amazonaws.com/sox-traffic-envoy-endpoint-register:86d3a57a51f62aa5a03cb66e6c64812e5e386e32;envoy-endpoint-register
2023-10-18 23:27:41,257 deployd.common.utils:255 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: name is envoy-endpoint-register
2023-10-18 23:27:41,257 deployd.common.utils:257 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: name is envoy-endpoint-register, status is b'healthy\n'
2023-10-18 23:27:41,257 deployd.common.utils:249 INFO for commit 86d3a57a51f62aa5a03cb66e6c64812e5e386e32: 998131032990.dkr.ecr.us-east-1.amazonaws.com/sox-traffic-envoy-sds:86d3a57a51f62aa5a03cb66e6c64812e5e386e32;envoy-sds
2023-10-18 23:27:41,271 deployd.common.utils:241 INFO Get health info for container with commit 636c069f9b8f7cada99e87eac77b2b8a04872a18
2023-10-18 23:27:41,289 deployd.common.utils:249 INFO for commit 636c069f9b8f7cada99e87eac77b2b8a04872a18: 998131032990.dkr.ecr.us-east-1.amazonaws.com/helloworlddummyservice:636c069f9b8f7cada99e87eac77b2b8a04872a18;helloworlddummyservice
2023-10-18 23:27:41,303 deployd.common.utils:241 INFO Get health info for container with commit e508d8c9bad67ee4403f818c912a6c24002136f1
2023-10-18 23:27:41,321 deployd.common.utils:249 INFO for commit e508d8c9bad67ee4403f818c912a6c24002136f1: 998131032990.dkr.ecr.us-east-1.amazonaws.com/mcrouter:e508d8c9bad67ee4403f818c912a6c24002136f1;mcrouter
2023-10-18 23:27:41,336 deployd.common.utils:255 INFO for commit e508d8c9bad67ee4403f818c912a6c24002136f1: name is mcrouter
2023-10-18 23:27:41,336 deployd.common.utils:257 INFO for commit e508d8c9bad67ee4403f818c912a6c24002136f1: name is mcrouter, status is b'healthy\n'
2023-10-18 23:27:41,336 deployd.common.utils:241 INFO Get health info for container with commit 8706351475e18caff1f1fdbd8e893e3ccb013bfa
2023-10-18 23:27:41,354 deployd.common.utils:249 INFO for commit 8706351475e18caff1f1fdbd8e893e3ccb013bfa: 998131032990.dkr.ecr.us-east-1.amazonaws.com/metrics-agent:8706351475e18caff1f1fdbd8e893e3ccb013bfa;metrics-agent
2023-10-18 23:27:41,368 deployd.common.utils:241 INFO Get health info for container with commit 34d94ff3700c1144052c69f14e4741a18e3805ab
2023-10-18 23:27:41,386 deployd.common.utils:249 INFO for commit 34d94ff3700c1144052c69f14e4741a18e3805ab: 998131032990.dkr.ecr.us-east-1.amazonaws.com/pastis_proxy:34d94ff3700c1144052c69f14e4741a18e3805ab;pastis_proxy
2023-10-18 23:27:41,401 deployd.common.utils:241 INFO Get health info for container with commit 946cba44b7b6c895a43e314a6ad8748c1221ddd1
2023-10-18 23:27:41,419 deployd.common.utils:249 INFO for commit 946cba44b7b6c895a43e314a6ad8748c1221ddd1: 998131032990.dkr.ecr.us-east-1.amazonaws.com/sox-traffic-ratelimit:946cba44b7b6c895a43e314a6ad8748c1221ddd1;ratelimit-A
2023-10-18 23:27:41,433 deployd.common.utils:255 INFO for commit 946cba44b7b6c895a43e314a6ad8748c1221ddd1: name is ratelimit-A
2023-10-18 23:27:41,433 deployd.common.utils:257 INFO for commit 946cba44b7b6c895a43e314a6ad8748c1221ddd1: name is ratelimit-A, status is b'healthy\n'
2023-10-18 23:27:41,433 deployd.common.utils:241 INFO Get health info for container with commit d5e571fc9694e746e133cc9ef860f0cb4ce9e474
2023-10-18 23:27:41,450 deployd.common.utils:249 INFO for commit d5e571fc9694e746e133cc9ef860f0cb4ce9e474: 998131032990.dkr.ecr.us-east-1.amazonaws.com/singer:d5e571fc9694e746e133cc9ef860f0cb4ce9e474;singer
2023-10-18 23:27:41,465 deployd.common.utils:241 INFO Get health info for container with commit abe3077cb493409e5265de35888389578d39ae9e
2023-10-18 23:27:41,484 deployd.common.utils:249 INFO for commit abe3077cb493409e5265de35888389578d39ae9e: 998131032990.dkr.ecr.us-east-1.amazonaws.com/tcollector:abe3077cb493409e5265de35888389578d39ae9e;tcollector
2023-10-18 23:27:41,500 deployd.common.utils:255 INFO for commit abe3077cb493409e5265de35888389578d39ae9e: name is tcollector
2023-10-18 23:27:41,501 deployd.common.utils:257 INFO for commit abe3077cb493409e5265de35888389578d39ae9e: name is tcollector, status is b'healthy\n'
2023-10-18 23:27:41,501 deployd.common.utils:241 INFO Get health info for container with commit d0eb5f6963ba397b96e39d59d0ab32ab29395d3f
2023-10-18 23:27:41,522 deployd.common.utils:241 INFO Get health info for container with commit 17e4ede82b4e69028a43478a33458efc45e59b03
2023-10-18 23:27:41,542 deployd.common.utils:249 INFO for commit 17e4ede82b4e69028a43478a33458efc45e59b03: 998131032990.dkr.ecr.us-east-1.amazonaws.com/zk-update-monitor:17e4ede82b4e69028a43478a33458efc45e59b03;zk_update_monitor
```

Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:227]: it is not easy to add UT for this method.
I can show you the database data if that is ok for you ^^

```
mysql> select * from agents where host_name="dev-yaqinli" and container_health_status!="";
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+--------------------------------------------------+
| host_name   | env_id                 | deploy_id              | deploy_stage  | state  | status    | start_date    | last_update   | last_operator | last_err_no | last_err_id | fail_count | first_deploy | host_id             | first_deploy_time | stage_start_date | container_health_status                          |
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+--------------------------------------------------+
| dev-yaqinli | 2XwshUOHR36DGEAH4PzBwg | 3zZuGjXbReqgsgwkAFcQPQ | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676388311 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676388311 | mcrouter:healthy;                                |
| dev-yaqinli | gu2DEmL4Tv2XOqDKFc7Axw | rXqFmaubTIGROjX0XT4VAg | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676388311 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676388311 | ratelimit-A:healthy;                             |
| dev-yaqinli | inbDfs6KQICpRRfuKQw_bQ | IS_sXYi4RyGfHAojyXllcw | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676388311 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676388311 | envoy-A:healthy;envoy-endpoint-register:healthy; |
| dev-yaqinli | pZwvW9faQTW7vD19ER72Gg | E8U5leLLSmmmAtkiqqy-kQ | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676388311 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676388311 | tcollector:healthy;                              |
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+--------------------------------------------------+
```

**Comment by liyaqin1** [comment]
Here are the result for my devapp:
```
mysql> select * from agents where host_name="dev-yaqinli" and container_health_status!="";
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+-------------------------------------------------+
| host_name   | env_id                 | deploy_id              | deploy_stage  | state  | status    | start_date    | last_update   | last_operator | last_err_no | last_err_id | fail_count | first_deploy | host_id             | first_deploy_time | stage_start_date | container_health_status                         |
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+-------------------------------------------------+
| dev-yaqinli | 2XwshUOHR36DGEAH4PzBwg | 3zZuGjXbReqgsgwkAFcQPQ | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676924048 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676924048 | mcrouter:healthy                                |
| dev-yaqinli | gu2DEmL4Tv2XOqDKFc7Axw | rXqFmaubTIGROjX0XT4VAg | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676924048 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676924048 | ratelimit-A:healthy                             |
| dev-yaqinli | inbDfs6KQICpRRfuKQw_bQ | IS_sXYi4RyGfHAojyXllcw | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676924048 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676924048 | envoy-A:healthy;envoy-endpoint-register:healthy |
| dev-yaqinli | pZwvW9faQTW7vD19ER72Gg | E8U5leLLSmmmAtkiqqy-kQ | SERVING_BUILD | DELETE | SUCCEEDED | 1691714754494 | 1697676924048 | system        |           0 | NULL        |          0 |            0 | i-0dd4b6ce5062c36f0 |     1691714754494 |    1697676924048 | tcollector:healthy                              |
+-------------+------------------------+------------------------+---------------+--------+-----------+---------------+---------------+---------------+-------------+-------------+------------+--------------+---------------------+-------------------+------------------+-------------------------------------------------+
4 rows in set (0.00 sec)


```

File: deploy-agent/deployd/common/utils.py [hunk 3]
```diff
@@ -219,18 +219,31 @@ def get_info_from_facter(keys):
...
+            lines = output.decode().strip().splitlines()
+            for line in lines:
+                if commit in line:
+                    parts = line.split(';')
+                    name = parts[1]
+                    try:
+                        command = ['docker', 'inspect', '-f', '{{.State.Health.Status}}', name]
+                        status = subprocess.run(command, check=True, stdout=subprocess.PIPE).stdout
+                        if status:
+                            result = result + name + ":" + status.decode().strip() + ";"
```
Comment (ntascii) [inline deploy-agent/deployd/common/utils.py:239]:
```
results = [] 
if status: results.append(f"{name}:{status.decode().strip()}")
return ";".join(results) if results else None
```

Comment (liyaqin1) [reply deploy-agent/deployd/common/utils.py:239]: cool, thank you ^^

**Review by ntascii — APPROVED** [review]

**Pull request merged**